contracts/farm/LiquidityMining.sol

```diff
@@ -731,4 +764,65 @@ contract LiquidityMining is ILiquidityMining, LiquidityMiningStorage {
         UserInfo memory ui = getUserInfo(_poolToken, _user);
         return ui.amount;
     }
+
+    /**
+     * @notice returns the accumulated liquid reward for the given user for each pool token
+     * @param _user the address of the user
+     */
+    function getUserAccumulatedLiquidReward(address _user)
+        external
+        view
+        returns (uint256[] memory)
+    {
+        uint256 length = poolInfoList.length;
+        uint256[] memory rewardList = new uint256[](length);
+        for (uint256 i = 0; i < length; i++) {
+            address _poolToken = address(poolInfoList[i].poolToken);
+            uint256 _unlockedImmediatelyPercent =
+                _getPoolTokenUnlockedImmediatelyPercent(_poolToken);
+            rewardList[i] =
+                (_unlockedImmediatelyPercent * _getUserAccumulatedReward(i, _user)) /
+                10000;
+        }
+
+        return rewardList;
+    }
+
+    /**
+     * @notice returns the accumulated vested reward for the given user for each pool token
+     * @param _user the address of the user
+     */
+    function getUserAccumulatedVestedReward(address _user)
+        external
+        view
+        returns (uint256[] memory)
+    {
```

Comment:
i suggest the following optimization:  
1) create an internal function to get accumulated rewards to be paid liquid
2) call this func from the external respectively for liquid rewards
3) the func to calc vested rewards is just a total amount accumulated sub liquid amount (p.1)

Reply:
IMO, the current one is more readable, rather than creating the internal function for liquid reward to be reuse it both liquid & vested reward exterrnal function.
Also, to calculate the `total amount accumulated`, it needs another loop, which means there will be 2 loops (1 for calculating the total amount accumulated, and the other one inside the internal function, so not really efficient.